### DISCUSSION TITLE
Flow Job Delay Not Working

### DISCUSSION THREAD

**Discussion by ccravens** [taskforcesh/bullmq#998] [Q&A] [answered]
Hey all!

Really enjoying working with BullMQ the past couple of months :) One issue I'm facing is I am not able to delay a job submitted to a flow. I have a flow where I'm adding about 7 parents to execute one after another. One of those workers I need to delay but about 30 secs. I'm doing that with the following:

```typescript
      {
        name: 'JOB_NAME',
        queueName: 'QUEUE_NAME',
       {},
        prefix: '',
       {
           delay: 30000
       },
        children: [],
      };
```

I am able to get a delay with Jobs that are not part of a flow. And I am creating a `QueueScheduler` for the Queues that I would like to delay with the following:

```typescript
new QueueScheduler('QUEUE_NAME', {connection: ...})
```

Any ideas what I may be missing? Thanks!

**Comment by manast** [comment]
This may be a missing feature actually. @roggervalf what do you think?

**Reply by roggervalf** [reply]
@manast probably we could check if the parent job has a delay to move it to delayed instead of wait when all children are completed if this is the expectation

**Reply by ccravens** [reply]
Hello @roggervalf, so for my particular use case the expectation would be that the job with the delay parameter would be delayed by the configured time starting once the child job (previous job) as completed.

Please let me know if there needs to be more clarification or discussion, thank you!

**Reply by manast** [reply]
> @manast probably we could check if the parent job has a delay to move it to delayed instead of wait when all children are completed if this is the expectation

Yes I think that is the expectation, it should be trivial to implement I think.

**Reply by ccravens** [reply]
Hello @manast! I wanted to check in and see if this fix has a planned release? I'm blocked on this and wanted to see if I should go ahead and create a work-around or wait for the feature to be released.

Thank you! :)

**Reply by manast** [reply]
@ccravens could you please create an issue with this bug/missing feature? as it is only a discussion now.

**Reply by ccravens** [reply]
@manast absolutely! Here you go :)

https://github.com/taskforcesh/bullmq/issues/1036

**Reply by ccravens** [reply]
@manast @roggervalf Hello! I wanted to check in on this issue to see if it had a planned release?

Thank you!

**Reply by roggervalf** [reply, marked as answer]
This feature is available since version 3.2.0
